### PR TITLE
Add subscription term to cloud router

### PR DESCRIPTION
**Pull request by vgvm-pf**
This update adds a subscription term to the cloud router resource. Valid values are 1, 12, 24, and 36, with a default value of 1. The subscription term is enforced at both the client and provider schema levels.

Specifically, this update makes changes to the following resources and data sources:

Data sources:
- cloud_connection
- cloud_connections
- cloud_routers

Resources:
- aws_cloud_router_connection
- azure_cloud_router_connection
- cloud_router
- cloud_router_connection_common
- cloud_router_quick_connect
- google_cloud_router_connection
- ibm_route_cloud_connection
- ipsec_route_cloud_connection
- oracle_cloud_router_connection
- outbound_cross_connect
- port_cloud_router_connection

This update also introduces the TF_VAR_pf_cs_oracle_region environment variable for Oracle testing. This variable should be set to the Oracle region that matches the specified TF_VAR_pf_cs_oracle_drg_ocid value.

Finally, two new methods are added to the client: FunctionRetry and Retry. These methods facilitate status polling mechanics, which can help to avoid some backend timing issues.

### CODE REVIEW

File: internal/provider/resource_azure_cloud_router_connection.go [hunk 1]
```diff
@@ -296,6 +306,9 @@ func extractAzureExpressRouteConn(d *schema.ResourceData) packetfabric.AzureExpr
 	if speed, ok := d.GetOk("speed"); ok {
 		expressRoute.Speed = speed.(string)
 	}
+	if subscriptionTerm, ok := d.GetOk("subscription_term"); ok {
+		expressRoute.SubscriptionTerm = subscriptionTerm.(int)
```
Comment (AndreasSteinerPF) [inline internal/provider/resource_azure_cloud_router_connection.go:310]: Nothing for this PR, but the naming in this whole file is off - expressRoute is something entirely different than an Azure Cloud Router Connection.

File: internal/provider/datasource_cloud_routers.go [hunk 2]
```diff
@@ -60,6 +60,11 @@ func dataSourceCloudRouters() *schema.Resource {
 							Type:     schema.TypeString,
 							Computed: true,
 						},
+						"subscription_term": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Subscription term of the Cloud Router Connection",
```
[the remark this answers is not in the document]
Comment (vgvm-pf) [reply internal/provider/datasource_cloud_routers.go]: Good catch! Fix it.

File: internal/provider/resource_azure_cloud_router_connection.go [hunk 3]
```diff
@@ -220,8 +228,10 @@ func resourceAzureExpressRouteConnRead(ctx context.Context, d *schema.ResourceDa
 	_ = d.Set("circuit_id", resp.CloudRouterCircuitID)
 	_ = d.Set("description", resp.Description)
 	_ = d.Set("speed", resp.Speed)
+	_ = d.Set("subscription_term", resp.SubscriptionTerm)
 	_ = d.Set("azure_service_key", resp.CloudSettings.AzureServiceKey)
 	_ = d.Set("po_number", resp.PONumber)
+	_ = d.Set("subscription_term", resp.SubscriptionTerm)
```
[the remark this answers is not in the document]
Comment (vgvm-pf) [reply internal/provider/resource_azure_cloud_router_connection.go]: Thanks! Removed it

File: internal/provider/resource_azure_cloud_router_connection.go (hunk 1 above)
Comment (vgvm-pf) [reply internal/provider/resource_azure_cloud_router_connection.go:309]: Yeah... there's definitely some of that throughout the provider.

**Pull request merged**